bare-metal-1.ks.in

```diff
@@ -0,0 +1,99 @@
+#version=DEVEL
+#test name: bare-metal-1
+
+# Use defaults for network, bootloader and storage.
+%ksappend network/default.ks
+%ksappend bootloader/default.ks
+%ksappend storage/default.ks
+
+# Install the prepared tar.
+liveimg --url=http://10.43.136.2/trees/rv/bmi/BP2-0.0.4.tar.xz
+
+# Run the installation in the cmdline mode.
+cmdline
+shutdown
+
+%post --nochroot
+
+# Download the tar image.
+tar_url="http://10.43.136.2/trees/rv/bmi/BP2-0.0.4.tar.xz"
+tar_img="/tmp/img.tar.xz"
+curl -f "${tar_url}" > "${tar_img}"
+
+%end
+
+%post --nochroot
+
+# Generate a list of files to exclude.
+cat > "/tmp/excluded.in" <<EOF
+./boot/
+./dev/
+./proc/
+./tmp/
+./sys/
+./run/
+./etc/machine-id
+./usr/lib/modules/
+EOF
+
+%end
+
+%post --nochroot
+
+# Find changed files.
+tar_img="/tmp/img.tar.xz"
+
+# Get a list of changed files.
+tar --diff -f "${tar_img}" -C /mnt/sysroot | cut -d":" -f1 \
+| grep -v -f /tmp/excluded.in| sort | uniq >/tmp/generated.out
+
+# Get a list of expected changes.
+cat > "/tmp/expected.out" <<EOF
+./etc/mtab
+./usr/libexec/openssh/ssh-keysign
+./var/lib/unbound/root.key
+EOF
+
+# Check the output
+diff /tmp/expected.out /tmp/generated.out
+
+if [[ $? != 0 ]]; then
+    echo "*** unexpected files has changed:" >> /mnt/sysroot/root/RESULT
+    cat /tmp/generated.out >> /mnt/sysroot/root/RESULT
+fi
```

Comment:
Shouldn't we also compare files directly. I mean if file in a tarball is really what we installed? It could be done by hashing or just `diff -q`.

Reply:
That was the plan, but I am not able to unpack the original tar image in stage2. I would have to use an additional storage and that would be painful. I had outputs from the diff as well and the results are the same.